---
title: 'Exercices'
description: 'meta description of the page'
---

## The dataset

Here, our dataset contains several informations related to transcripts in the human genome. They were computed using *pygtftk* (v0.9.8) from a GTF file downloaded from ensembl (genome version GRCh38, release 92). First we will load the dataset, set the row names to the transcript ids and inspect the column names.

```r
# Only in the webpage
webr::install("knitr")
webr::install("data.table")
webr::install("dplyr")
webr::install("curl")

library(dplyr)
library(knitr)
library(data.table)

# In your personal R
# if(!require(dplyr)){install.packages("dplyr")}
# if(!require(data.table)){install.packages("data.table")}
# if(!require(knitr)){install.packages("knitr")}

tx_info <- as.data.frame(fread("https://tinyurl.com/hg38-tx-info-txt", sep="\t", header = T))
rownames(tx_info)  <- tx_info[,11]
kable(colnames(tx_info))
```

First, we will only select transcript belonging to a class of *gene_biotype* which contains more than 500 transcripts.

```r
tx_info <- tx_info %>%
            group_by(gene_biotype) %>%
            filter(n() > 500)
```

Then we will use an ordered factor for chromosomes:

```r
tx_info$seqid <- factor(tx_info$seqid, ordered = T, levels = c(paste("chr", 1:22, sep=""), "chrX", "chrY"))
```

::Exo
---
consigne: "Create a faceted histogram showing the distribution of transcript genomic size"
solution: |
  ggplot(tx_info, aes(x=log10(tx_genomic_size))) + 
  geom_histogram(fill="#E69F00", color="white") + 
  theme(axis.text.x = element_text(angle = 45))
---
::

::Exo
---
consigne: "Create a faceted histogram showing the distribution of transcript genomic size depending on gene_biotype"
solution: |
  ggplot(tx_info, aes(x=log10(tx_genomic_size))) + 
  geom_density(fill="#E69F00", color="white") + 
  theme(axis.text.x = element_text(angle = 45)) +
  facet_grid(gene_biotype~., scales  = "free_y") + 
  theme(strip.text.y = element_text(angle = 0))
---
::

::Exo
---
consigne: "Draw a stacked barplot showing the number of transcripts from each gene_biotype class on each chromosome"
solution: |
  ggplot(tx_info, aes(x=seqid, fill=gene_biotype)) + 
  geom_bar(aes(fill=gene_biotype),  color="white") +
  theme(axis.text.x = element_text(angle = 45, hjust=1)) +
  theme(strip.text.x = element_text(size = 7, 
                                    colour = 'black',
                                    angle = 90),
        legend.position = "bottom")
---
::
